pallets/connectors/src/inbound.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_traits::{
+	ops::{EnsureAdd, EnsureSub},
+	CurrencyInspect, Investment, InvestmentCollector, Permissions,
+};
+use cfg_types::{
+	domain_address::{Domain, DomainAddress},
+	permissions::{PermissionScope, PoolRole, Role},
+};
+use frame_support::{
+	ensure,
+	traits::fungibles::{Mutate, Transfer},
+};
+use sp_runtime::{
+	traits::{Convert, Zero},
+	DispatchResult,
+};
+
+use crate::{
+	pallet::Error, Config, CurrencyIdOf, GeneralCurrencyIndexOf, Pallet, PoolIdOf, TrancheIdOf,
+};
+
+impl<T: Config> Pallet<T> {
+	/// Executes a transfer from another domain exclusively for
+	/// non-tranche-tokens.
+	///
+	/// Directly mints the currency into the receiver address.
+	pub fn do_transfer_from_other_domain(
+		currency: GeneralCurrencyIndexOf<T>,
+		receiver: T::AccountId,
+		amount: <T as Config>::Balance,
+	) -> DispatchResult {
+		ensure!(!amount.is_zero(), Error::<T>::InvalidTransferAmount);
+
+		let currency_id = Self::try_get_currency_id(currency)?;
+		T::Tokens::mint_into(currency_id, &receiver, amount)?;
+
+		Ok(())
+	}
+
+	/// Executes a transfer from the `DomainLocator` account of the origination
+	/// domain to the receiver exclusively for tranche tokens.
+	///
+	/// Assumes that the amount of tranche tokens has been minted in the
```

Comment:
Tranche tokens are always minted by investments, only transferred into the `DomainLocator` account on `connectors.Transfer`
```suggestion
	/// Assumes that the amount of tranche tokens has been locked in the
```